File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.kt
class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        var wMax = 0
        var hMax = 0
        for(s in sizes){
            wMax = maxOf(wMax, s[0], s[1])
            hMax = Math.max(hMax, if(s[0]<s[1]) s[0] else s[1])
        }
        
        
        return hMax * wMax
    }
}

File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    #회전 : 가로 세로 길이를 교환
    w = []
    h = []
    for i in range(len(sizes)):
        if sizes[i][0] >= sizes[i][1]:
            w.append(sizes[i][0])
            h.append(sizes[i][1])
        else:
            w.append(sizes[i][1])
            h.append(sizes[i][0])
            
    
    return max(h) * max(w)